##  创建分析用的虚拟环境
```python
conda create -n sciRNAseq python3.7
conda activate sciRNAseq
pip install pandas numpy mappy gzip 
```

## 合并多个文件为一个文件

input_folder="/root/wangje/Project/吴霞/BCPB230608/Raw_Data" 
output_folder="/root/wangje/Project/吴霞/BCPB230608/output/01" 
ligation_barcode_file="/root/wangje/Project/吴霞/script/ligation_barcode.txt"
RT_Barcode_file="/root/wangje/Project/吴霞/script/RT_barcode.txt"
p7_file="/root/wangje/Project/吴霞/BCPB230608/script/Wu_sciRNA-seq/script/sample_p7.txt"
id=BCPB230608
for i in {65..96}
do 
    sample_id="${id}_${id}-${i}"
    echo "$sample_id"
    python /root/wangje/Project/吴霞/BCPB230608/script/Wu_sciRNA-seq/script/01_根据umi提取符合条件的barcode并输出文件.py ${input_folder} $output_folder $sample_id $ligation_barcode_file $RT_Barcode_file $p7_file
done

# trim
input_folder=/root/wangje/Project/吴霞/BCPB230608/output/01
output_folder=/root/wangje/Project/吴霞/BCPB230608/output/02
for i in {61..96}
do 
    trim_galore $input_folder/BCPB230608_BCPB230608-${i}_R2_barcode.fq.gz  -a AAAAAAAA --three_prime_clip_R1 1 -o $output_folder
done

# STAR

core=10
index="/root/wangje/Reference/Star_index/Human/GeneCode/"
STAR_output_folder="/root/wangje/Project/吴霞/BCPB230608/output"
input_folder="/root/wangje/Project/吴霞/BCPB230608/output/02"

for i in {61..96}
do
STAR \
    --runThreadN $core \
    --outSAMstrandField intronMotif \
    --genomeDir $index --readFilesCommand zcat \
    --readFilesIn $input_folder/BCPB230608_BCPB230608-${i}_R2_barcode_trimmed.fq.gz\
    --outFileNamePrefix $STAR_output_folder/BCPB230608-${i}- \
    --genomeLoad LoadAndKeep
done

input_folder="/root/wangje/Project/吴霞/BCPB230608/output/03"
output_folder="/root/wangje/Project/吴霞/BCPB230608/output/04"
sample_id=BCPB230608
for i in {1..96}
do
    echo ">>>>>>>>>> sample: ${sample_id}-${i}"
    samtools view -bh -q 30 -F 4 $input_folder/${sample_id}-${i}-Aligned.out.sam|samtools sort -@ 10 -|samtools view -h ->$output_folder/${sample_id}-${i}_filterAndSort.sam
done

# Remove Duplicate UMI
input_folder="/root/wangje/Project/吴霞/BCPB230608/output/04"
sample_prefix="BCPB230608"
output_folder="/root/wangje/Project/吴霞/BCPB230608/output/05"

for i in {1..96}
do 
    python /root/wangje/Project/吴霞/BCPB230608/script/Wu_sciRNA-seq/script/05_removeDuplicateUmi.py ${input_folder} ${sample_prefix}-${i}_filterAndSort ${output_folder}
done

# split sam by barcode
ulimit -n 100000
# sam_file="/home/data/vip9t22/Project/BC2305/BC230502-" 
# barcode_file="/home/data/vip9t22/Project/BC2305/combined_barcode.txt" 
# output_folder="/home/data/vip9t22/Project/BC2305/BC230502-"

sam_file="/root/wangje/Project/吴霞/BCPB230608/output/05/BCPB230608-"
barcode_file="/root/wangje/Project/吴霞/BCPB230608/script/Wu_sciRNA-seq/script/combined_barcode.txt"
output_folder="/root/wangje/Project/吴霞/BCPB230608/output/06/BCPB230608-"
cutoff=3
for i in {1..96}
do 
    echo "BCPB230608-${i}"
    python /root/wangje/Project/吴霞/script/06_splitSAM.py ${sam_file}${i}_filterAndSort_rmDup.sam ${barcode_file} ${output_folder}${i} $cutoff
done

# gene计数
gtf_file="/root/wangje/Reference/Homo_sapiens/GeneCode/hg38/Annotation/Genes/gencode.v43.chr_patch_hapl_scaff.annotation.gtf"
# input_folder="/root/wangje/Project/吴霞/Data/06_splitSAM"
input_folder="/root/wangje/Project/吴霞/BCPB230608/output/06"
# sample_ID="/root/wangje/Project/吴霞/Data/06_splitSAM"
sample_ID="/root/wangje/Project/吴霞/BCPB230608/output/06"
core=40

for i in {30..59}
do 
    python  /root/wangje/Project/吴霞/script/07.py ${gtf_file} ${input_folder}/BCPB230608-${i} ${sample_ID}/BCPB230608-${i}/BCPB230608-${i}_filterAndSort_rmDup.sample_list.txt ${core}
done
